third_party/iap_jwt_verify_nginx/src/iap_jwt_verification.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2002-2016 Igor Sysoev
+// Copyright (C) 2011-2016 Nginx, Inc.
+// Copyright (C) 2017 Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+// ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+// SUCH DAMAGE.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "src/iap_jwt_verification.h"
+
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "openssl/bn.h"
+#include "openssl/bytestring.h"
+#include "openssl/ecdsa.h"
+#include "openssl/sha.h"
+#include "src/jwt.h"
+#include "src/key_util.h"
+
+namespace google {
+namespace cloud {
+namespace iap {
+
+// Let's refuse to even process JWTs larger than a certain size.
+constexpr size_t MAX_RAW_JWT_SIZE = 4096;
```

Comment:
Yep, I checked the typical length first...this is safely bigger than any JWT we're likely to issue even if we increase our key size or add more fields to the JWT, but it saves us the stomach ulcers of wondering "What happens if someone malicious sends us a JWT the size of the moon?".